modules/storages/app/controllers/storages/admin/storages_controller.rb

```diff
@@ -0,0 +1,163 @@
+  # See https://guides.rubyonrails.org/layouts_and_rendering.html for reference on layout
+  layout 'admin'
+
+  # specify which model #find_model_object should look up
+  model_object Storages::Storage
+
+  # Before executing any action below: Make sure the current user is an admin
+  # and set the @<controller_name> variable to the object referenced in the URL.
+  before_action :require_admin
+  before_action :find_model_object, only: %i[show destroy edit update]
+
+  # menu_item is defined in the Redmine::MenuManager::MenuController
+  # module, included from ApplicationController.
+  # The menu item is defined in the engine.rb
+  menu_item :storages_admin_settings
+
+  # Index page with a list of Storages objects
+  # Called by: Global app/config/routes.rb to serve Web page
+  def index
+    @storages = Storages::Storage.all
+  end
+
+  # Show page with details of one Storage object.
+  # Called by: Global app/config/routes.rb to serve Web page
+  def show; end
+
+  # Show the admin page to create a new Storage object.
+  # Sets the attributes provider_type and name as default values and then
+  # renders the new page (allowing the user to overwrite these values and to
+  # fill in other attributes).
+  # Used by: The index page above, when the user presses the (+) button.
+  # Called by: Global app/config/routes.rb to serve Web page
+  def new
+    # Set default parameters using a "service".
+    # See also: storages/services/storages/storages/set_attributes_services.rb
+    # See also: https://www.openproject.org/docs/development/concepts/contracted-services/
+    # That service inherits from ::BaseServices::SetAttributes
+    @object = ::Storages::Storages::SetAttributesService
+                .new(user: current_user,
+                     model: Storages::Storage.new,
+                     contract_class: EmptyContract)
+                .call(provider_type: 'nextcloud', name: I18n.t('storages.provider_types.nextcloud'))
+                .result
+  end
+
+  # Actually create a Storage object.
+  # Overwrite the creator_id with the current_user. Is this this pattern always used?
+  # Use service pattern to create a new Storage
+  # See also: storages/services/storages/storages/create_service.rb
+  # See also: https://www.openproject.org/docs/development/concepts/contracted-services/
+  # Called by: Global app/config/routes.rb to serve Web page
+  # rubocop:disable Metrics/AbcSize
+  def create
+    combined_params = permitted_storage_params.to_h.reverse_merge(creator_id: current_user.id)
+    service_result = Storages::Storages::CreateService.new(user: current_user).call(combined_params)
+    @object = service_result.result
+
+    if service_result.success?
+      flash[:notice] = I18n.t(:notice_successful_create)
+      # storage_path is automagically created by Ruby routes.
+      redirect_to storage_path(@object)
+    else
+      @errors = service_result.errors
+      render :new
+    end
+  end
```

Comment:
Again, the `creator_id: current_user` part can be placed inside the `SetAttributesService` which will reduce the AbcSize as a side effect.

Reply:
Done in 9bc93a91, and it did remove the AbcSize warning :+1: .